doc/rm/comportability_specification/index.md

```diff
@@ -440,18 +440,70 @@ Unlike interrupts, there is no software component to alerts at the peripheral, t
+ECC uncorrectable in Flash                                 | x           | x                 |
+ECC uncorrectable in OTP                                   | (x)         |                   | x
+Any ECC / Parity error in SRAMs or register files          | (x)         |                   | x
+Glitch detectors (e.g., invalid FSM encoding)              | (x)         |                   | x
+Incorrect usage of security IP (e.g., AES config register) | x           | x                 |
+Incorrect usage of regular IP                              | x           |                   |
+
+(x): optional
+
+The column "Regular IRQ" indicates whether the corresponding error condition should also send out a regular IRQ.
+Fatal alerts are not required to send out regular IRQs, since they cannot be recovered via SW actions.
+
+### Defining Alerts
+
+The Hjson configuration file defined above specifies all that needs to be known about the alerts in the standard case.
+The following sections specify what comes out of various tools based upon the simple list defined in the above example.
+
+In terms of naming convention, recoverable alerts must be prefixed with `recov_*`, whereas fatal alerts must be prefixed with `fatal_*`.
+If a module only outputs one alert of each type, it is permissible to name the alerts `recoverable`/`recov` and/or `fatal`.
+
+### Test Alert Register Creation
+
+For every peripheral, by default, one register named `ALERT_TEST` is **automatically** created.
+This can be overridden within the `reggen` tool by specifying `no_auto_alert_regs = true`.
+
+`ALERT_TEST` is a write-only (`wo`) register that allows software to test the reporting of alerts in the alert handler.
+Every alert of a peripheral has one field bit inside the `ALERT_TEST` register, and each field bit is meant to be connected to the test input of the corresponding `prim_alert_sender` (see next subsection).
+
 ### Alert Hardware Implementation
 
 Internal events are sent active-high to a piece of IP within the peripheral called the `prim_alert_sender`.
-One `prim_alert_sender` must be instantiated per distinct alert event type.
+One `prim_alert_sender` must be instantiated per distinct alert event, and the `IsFatal` parameter of the alert sender must be set to 1 for fatal alerts (this causes the alert sender to latch the alert until the next system reset).
+
 It is up to the peripheral owner to determine what are distinct alert events;
 multiple ones can be bundled depending upon the distinction required within the module (i.e.  high priority threat vs. low level threat).
+However, it is recommended that comportable IPs with multiple bundled alerts expose an error register to indicate the cause.
+Error registers for recoverable alerts must either be clearable by SW, or the HW must provide an automatic mechanism to clear them (e.g., upon starting a new transaction initiated by SW).
+Error registers for fatal alerts must not be clearable in any way and must hence be read-only.
```

Comment:
Rather than make this optional, it will be useful (for automating DV testing) to always enforce this. Perhaps automatically also add `ALERT_STATE` CSR?

Reply:
I think the error registers that Michael's talking about are for disambiguating within an alert. For example, events FOO and BAR both send alert BAZ; this register lets e.g. the host processor work out what's going on. This can't be automated, because it conveys information that isn't part of the generic alert infrastructure.

@msfschaffner: Maybe this would be clearer if you changed "error register" to something like "cause register" or "alert cause register"?

Reply:
I think @rswarbrick is right: this is more about an alert cause register.

I like the idea of an `ALERT_STATE` CSR as it can help for automated CSR testing. However, considering that most peripherals will just have two alerts (`recov`/`fatal`) of which one is always asserted for just one cycle and thus not observable for SW (`recov`) there is little value. In AES we currently have two status bits to indicate if there was an alert. The `recov` bit remains high until the main control register is rewritten (that how we define to recover from such an alert: restart AES operation).

Regarding the cause register: Maybe it's worth to add a comment that exposing the error cause can be sensitive and that reads to those registers should potentially guarded by LC.

Reply:
I see, nvm then.

Reply:
Yes, the registers I am talking about are cause registers that would allow SW to figure out the root cause in case multiple alert sources have been collapsed into one alert channel, and hence this is specific to each IP.

Adding alert state registers would be possible, but as @vogelpi mentioned, there is likely little value in doing so.

Regarding LC gating: before putting this into our guidance, we need consultancy input. I am about to add this question to our tracking doc, and will come back to this once we know how to proceed here.